pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -41,26 +45,91 @@ var log = logf.Log.WithName("DeploymentReconciler")
 
 // DeploymentReconciler reconciles the raw kubernetes deployment resource
 type DeploymentReconciler struct {
-	client       kclient.Client
-	scheme       *runtime.Scheme
-	Deployment   *appsv1.Deployment
-	componentExt *v1beta1.ComponentExtensionSpec
+	client         kclient.Client
+	scheme         *runtime.Scheme
+	DeploymentList []*appsv1.Deployment
+	componentExt   *v1beta1.ComponentExtensionSpec
 }
 
 func NewDeploymentReconciler(client kclient.Client,
 	scheme *runtime.Scheme,
 	componentMeta metav1.ObjectMeta,
+	workerComponentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
-	podSpec *corev1.PodSpec) *DeploymentReconciler {
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) *DeploymentReconciler {
 	return &DeploymentReconciler{
-		client:       client,
-		scheme:       scheme,
-		Deployment:   createRawDeployment(componentMeta, componentExt, podSpec),
-		componentExt: componentExt,
+		client:         client,
+		scheme:         scheme,
+		DeploymentList: createRawDeployment(componentMeta, workerComponentMeta, componentExt, podSpec, workerPodSpec),
+		componentExt:   componentExt,
 	}
 }
+func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec, //nolint:unparam
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) []*appsv1.Deployment {
+	var deploymentList []*appsv1.Deployment
+	var workerNodeSize string
+	var pipelineParallelSize string
+
+	isvcName := componentMeta.GetLabels()[constants.InferenceServiceLabel]
+	multiNodeEnabled := false
+
+	if workerComponentMeta.Name != "" {
```

Comment:
at some places you check if the `workerComponentMeta.Name` field is not empty and on another you check if the workerPodSpec is not nil, maybe it would be better if we define only one method that can be done to check it?

Reply:
This is because the method of verification differs depending on the file.

In the servingruntime, the most accurate way to check is whether workerSpec is nil. However, in deployment_reconciler, since the servingruntime object is not available, we need to check in a different way. The most accurate method in this case is to verify that the Name within workerComponentMeta is absent, as workerComponentMeta itself may not be nil.

If there is a universal method that can be applied across all contexts, I would appreciate your recommendation.